File: src/main/java/org/snake/views/SnakeNodeView.java
package org.snake.views;

import org.snake.models.SnakeNode;

import java.awt.*;

public class SnakeNodeView implements FigureView {
    private static final Color SNAKE_NODE_COLOR = new Color(105, 210, 0);

    private final SnakeNode model;

    public SnakeNodeView(SnakeNode model) {
        this.model = model;
    }

    public SnakeNode getModel() {
        return model;
    }

    @Override
    public boolean isEmptySpace() {
        return false;
    }

    @Override
    public void draw(Graphics canvas, int x, int y, int width, int height) {
        canvas.setColor(SNAKE_NODE_COLOR);
        canvas.fillOval(x, y, width, height);
        canvas.setColor(Color.BLACK);
        canvas.drawOval(x, y, width, height);

        if(model.isHead()){
            canvas.setColor(Color.BLUE);
            canvas.fillOval(x + (width)/4, y + height/4, width/2, height/2);
            canvas.fillOval(x +(width * 3)/4, y + height/4, width/2, height/2);
        }
    }

}

File: src/main/kotlin/org/snake/views/SnakeNodeView.kt
package org.snake.views

import org.snake.models.SnakeNode
import java.awt.Color
import java.awt.Graphics

class SnakeNodeView(val model: SnakeNode) : FigureView {
    override val isEmptySpace: Boolean
        get() = false

    override fun draw(canvas: Graphics, x: Int, y: Int, width: Int, height: Int) {
        canvas.color = SNAKE_NODE_COLOR
        canvas.fillOval(x, y, width, height)
        canvas.color = Color.BLACK
        canvas.drawOval(x, y, width, height)
        if (model.isHead) {
            canvas.color = Color.BLUE
            canvas.fillOval(x + width / 4, y + height / 4, width / 2, height / 2)
            canvas.fillOval(x + width * 3 / 4, y + height / 4, width / 2, height / 2)
        }
    }

}